src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1565 @@
+                self.counter2)  # Check late subscriber
+
+    def test_get_and_subscribe(self):
+        field_images = (3, 4)
+        global counter, counter2
+        self.counter = 0
+        self.counter2 = 0
+
+        dataflow = self.MPPC.data
+        dataflow.subscribe(self.image_received)
+
+        for x in range(field_images[0]):
+            for y in range(field_images[1]):
+                dataflow.next((x, y))
+
+        # Acquire single field without unsubscribing listener (expect error)
+        with self.assertRaises(Exception):
+            image = dataflow.get()
+            self.assertIsInstance(image, model.DataArray)
+
+        time.sleep(field_images[0] * field_images[1]) # Wait a bit to allow some processing and receive images.
+        self.assertEqual(field_images[0] * field_images[1], self.counter)
+        dataflow.unsubscribe(self.image_received)
+
+        # Acquire single field after unsubscribing listener
+        image = dataflow.get()
+        self.assertIsInstance(image, model.DataArray)
+
+        # Start acquiring second mega field
+        dataflow.subscribe(self.image_2_received)
+        for x in range(field_images[0]):
+            for y in range(field_images[1]):
+                dataflow.next((x, y))
+
+        time.sleep(field_images[0] * field_images[1]) # Wait a bit to allow some processing and receive images.
+        dataflow.unsubscribe(self.image_2_received)
+        time.sleep(0.5)
+        self.assertEqual(field_images[0] * field_images[1], self.counter)
+        self.assertEqual(field_images[0] * field_images[1], self.counter2)
+
+class Test_AsmApiException(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW:
+            raise unittest.SkipTest('No HW or simulator for the ASM and HwComponents present. Skipping tests.')
+
+    @classmethod
+    def tearDownClass(cls):
+        pass
+
+    def setUp(self):
+        self.ASM_manager = AcquisitionServer("ASM", "main", URL, children=CHILDREN_ASM, externalStorage=EXTRNAL_STORAGE)
+        for child in self.ASM_manager.children.value:
+            if child.name == CONFIG_MPPC["name"]:
+                self.MPPC = child
+            elif child.name == CONFIG_SCANNER["name"]:
+                self.EBeamScanner = child
+            elif child.name == CONFIG_DESCANNER["name"]:
+                self.MirrorDescanner = child
+
+        # Ensure that only empty images will be received
+        self.MPPC.dataContent.value = "empty"
+
+        resp = self.ASM_manager._session.get(self.ASM_manager._host + "/scan/clock_frequency",
+                                             json=None, timeout=600)
+
+        self.error_object = AsmApiException("test_url", resp, 404)
+
+    def tearDown(self):
+        self.ASM_manager.terminate()
+        time.sleep(0.2)
+
+    def test_AsmApiException(self):
+        with self.assertRaises(Exception):
+            raise self.error_object
```

Comment:
That's a gigantic amount of code for testing that `raise ` raises an execption!
I'd suggest to just add a short test in the AquisitionServer test that looks like:
```python
with self.assertRaises(AsmApiException):
    self.ASM_manager.asmApiGetCall(/fake/function", ...)
```